volatility3/framework/symbols/windows/extensions/registry.py

```diff
@@ -132,6 +173,103 @@ def get_full_key_name(self) -> str:
+                raise ValueError(
+                    f"Unable to read inline registry value with excessive length: {datalen}"
+                )
+            else:
+                data = layer.read(self.Data.vol.offset, datalen)
+        elif layer.hive.Version == 5 and datalen > 0x4000:
+            # We're bigdata
+            big_data = layer.get_node(self.Data)
+            # Oddly, we get a list of addresses, at which are addresses, which then point to data blocks
+            for i in range(big_data.Count):
+                # The value 4 should actually be unsigned-int.size, but since it's a file format that shouldn't change
+                # the direct value 4 can be used instead
+                block_offset = layer.get_cell(big_data.List + (i * 4)).cast(
+                    "unsigned int"
+                )
+                if (
+                    isinstance(block_offset, int)
+                    and block_offset < layer.maximum_address
+                ):
+                    amount = min(BIG_DATA_MAXLEN, datalen)
+                    data += layer.read(
+                        offset=layer.get_cell(block_offset).vol.offset, length=amount
+                    )
+                    datalen -= amount
+        else:
+            # Suspect Data actually points to a Cell,
+            # but the length at the start could be negative so just adding 4 to jump past it
+            data = layer.read(self.Data + 4, datalen)
+
+        if self.get_type() == RegValueTypes.REG_DWORD:
+            if len(data) != struct.calcsize("<L"):
+                raise ValueError(
+                    f"Size of data does not match the type of registry value {self.get_name()}"
+                )
+            (res,) = struct.unpack("<L", data)
+            return res
+        if self.get_type() == RegValueTypes.REG_DWORD_BIG_ENDIAN:
+            if len(data) != struct.calcsize(">L"):
+                raise ValueError(
+                    f"Size of data does not match the type of registry value {self.get_name()}"
+                )
+            (res,) = struct.unpack(">L", data)
+            return res
+        if self.get_type() == RegValueTypes.REG_QWORD:
+            if len(data) != struct.calcsize("<Q"):
+                raise ValueError(
+                    f"Size of data does not match the type of registry value {self.get_name()}"
+                )
+            (res,) = struct.unpack("<Q", data)
+            return res
+        if self.get_type() in [
+            RegValueTypes.REG_SZ,
+            RegValueTypes.REG_EXPAND_SZ,
+            RegValueTypes.REG_LINK,
+            RegValueTypes.REG_MULTI_SZ,
+            RegValueTypes.REG_BINARY,
+            RegValueTypes.REG_FULL_RESOURCE_DESCRIPTOR,
+            RegValueTypes.REG_RESOURCE_LIST,
+            RegValueTypes.REG_RESOURCE_REQUIREMENTS_LIST,
+        ]:
+            return data
+        if self.get_type() == RegValueTypes.REG_NONE:
+            return b""
```

Comment:
I'm used to seeing `REG_NONE` values being treated as empty binary containers, so I chose this return type as it was just consistent with my prior experience; I just did a sanity check across a couple of test hives to make sure that this will always be the case. I think we can safely return `None` here, I'll make the change.